398/V2 Footer component and styling

OlenaReukova opened the pull request:
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review on my code
- [ ] I have made corresponding changes to the documentation (if any were needed)
- [ ] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and previously existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description

**Relates #398 <u>OR</u> Closes #398**  
Include here a short description of what behaviour this Pull Request introduces.
Outline any new dependencies introduced.

### Files changed

Include one to two lines on the necessary changes to each file.

### UI changes

If this PR includes changes to the UI include a **screengrab** of the alterations in review.

### Changes to Documentation

Include an outline of why Documentation needed to be updated with this PR

# Tests

Include a description of any tests added - if none have been added explain why.

OlenaReukova committed:
added social icons Facebook, Instagram, Threads

OlenaReukova committed:
Merge branch 'dev' into 398-footer-component-and-styling
Merge dev branch to update footer component# the commit.

OlenaReukova committed:
refactor: remove width and height props from Social Icons

Simplified the component props by keeping only className.

OlenaReukova committed:
feat: add Footer component with company info and social links

Created a responsive Footer component with structured sections.
Displays company name, registration number, and address.
Includes social media links for Facebook, Instagram, and Threads.
Provides navigation links for company info, pages, and policies.
Adapts layout for mobile and desktop views .

OlenaReukova committed:
fix: correct Footer import path

OlenaReukova committed:
fix: correct Footer import path in Layout

OlenaReukova commented:
@nichgalzin Hi.
**What I have done so far:**

- Added social icons.
- Created a new Footer component.
- Imported the new Footer component into the page and layout.
- Fully responsive.

**What I did not do:**

- I did not add the configuration settings from Tailwind CSS because the updated version has not yet been merged.
- Cypress Test
- Remove the previous Footer component.
- Add actual links to all elements of the footer, as, according to a conversation with the UI/UX design team, they will be provided at a later stage of the design development.

nichgalzin commented:
Hey Olena, 
   I think go ahead and delete the old footer, I think it's a good idea in this footer to replace and remove anything that is not used anymore and just make sure to check all the places it's being used.
 As for cypress tests, lets at least make sure that we're not making any existing tests fail and that we create a storybook file for any new xu/ui components we create. Thanks!

OlenaReukova committed:
refactor:removed previos v1 Footer component

OlenaReukova committed:
refactor:replace hardcoded HEX color with Tailwind class (bg-brand-80, bg-primaryOrange )

OlenaReukova committed:
refactor: edit links to the Pages section (Browse Items, Post an Item, Messages)

OlenaReukova committed:
test: add test to check Footer rendering

nichgalzin commented on components/footer/Footer.tsx:
```diff
@@ -0,0 +1,173 @@
+    address: '7 Bell Yard, London, WC2A 2JR',
+  },
+  socialLinks = {
+    facebook: 'https://facebook.com',
```
Let's ask Katia for the actual links to the Trafalgar girls social media pages and use them instead?

OlenaReukova replied:
updated actual social media links

nichgalzin commented on components/icons/ThreadsIcon.tsx:
```diff
@@ -0,0 +1,25 @@
+    >
+      <g clipPath='url(#clip0)'>
+        <path
+          d='M18.75 7.61294C17.7994 5.14169 15.8034 3.33325 12 3.33325C6 3.33325 4.5 7.83325 4.5 12.3333C4.5 16.8333 6 21.3333 12 21.3333C16.5 21.3333 18.75 18.3333 18.75 16.0833C18.75 10.0833 9 10.0833 9 14.5833C9 18.3333 15.75 18.3333 15.75 12.3333C15.75 7.08325 10.5 7.08325 9 9.33325'
```
This looks like it needs to be formatted?

OlenaReukova replied:
refactored the paths for social media icons -done

nichgalzin commented on components/icons/socialIcons.ts:
```diff
@@ -0,0 +1,3 @@
+export { default as FacebookIcon } from './FacebookIcon';
```
We haven't been using this convention of have files dedicated to exporting. I'm not opposed to it but if this case can you make a directory called something like social-icons and then have all the component files and this in there?

OlenaReukova replied:
created social-icon folder and moved icons inside to keep all component files organised

nichgalzin commented on components/footer/Footer.tsx:
```diff
@@ -0,0 +1,173 @@
+
+const Footer: React.FC<FooterProps> = ({
+  companyInfo = {
+    name: 'KYNDLY CIC',
```
Passing the hard coded values as props is fine, but I think we should not set the values as defaults here. Lets make a separate file with these objects, we should have a directory called constants with these sorts of files in them, we have a few other files like this that we need to move there, so go ahead and set that up for us.

OlenaReukova replied:
1. Moved hardcoded footer constants to 'constants/footer.ts' for better reusability.
2. Extracted 'FooterProps' type to 'types/footer.ts' to keep concerns separate.
3. Updated 'Footer.tsx' to import constants and types from new locations.
This refactor improves code organisation, makes updates and maintain easier.

nichgalzin reviewed (changes requested):
Hi Oleana, great work this is looking good, just have a few notes:
- it seems like one the E2E tests is failing can you have a look at it and see what's wrong?
- the bg-color property is not working as you haven't pulled the latest changes that have now been merge into main. Can you make sure this branch is up to date before we merge it?
-On the desktop version the padding needs a bit of work, make sure there are even spacing of both sides, this it what it looks like to me.
![Screenshot 2025-03-03 at 16 28 24](https://github.com/user-attachments/assets/80b705f3-0316-43b3-8ed5-e534cec557eb)

Thanks again for your work!
_____________________________
@nichgalzin 
updated info: 1) The Cypress tests are failing because cy.click() is being called on multiple elements instead of a single one.

Current errors in 'Anonymous Navigation':
✅ Can see recently added items (2188ms) - passed
❌ 1) Can browse items
❌ 2) Can see about page
❌ 3) Add item redirects to login
❌ 4) Message redirects to login
❌ 5) Profile redirects to login"
2) I merged the updated dev branch into the current branch, and now we don't have an issue with the bg-color property.
3) I fixed the alignment of the footer section by using flex and justify-between. Now, it looks centred with accurate right and left padding.

OlenaReukova committed:
refactor: improve footer maintainability by separating constants and types

1. Moved hardcoded footer constants to 'constants/footer.ts' for better reusability.
2. Extracted 'FooterProps' type to 'types/footer.ts' to keep concerns separate.
3. Updated 'Footer.tsx' to import constants and types from new locations.
This refactor improves code organisation, makes updates and maintain  easier.

OlenaReukova committed:
refactor: update actual social media links

OlenaReukova committed:
Merge branch 'dev' into 398-footer-component-and-styling

OlenaReukova committed:
refactor: formatted paths for facebook, instagram, and threads icons

OlenaReukova committed:
refactor: fix footer positioning with mt-auto and min-h-[100dvh]

OlenaReukova committed:
refactor: add flex and justify-between to align items along the container's main axis, ensuring equal spacing between each item

OlenaReukova committed:
refactor: create social-icon folder and move icons inside to keep all component files organised

OlenaReukova committed:
refactor: add LinkedIn icon to the footer

OlenaReukova committed:
fix: adjusting the LinkedIn icon viewBox by reducing the height

OlenaReukova committed:
fix:mobile view

OlenaReukova committed:
test:add Check if the social media section is present

OlenaReukova committed:
test: add Check if the Company Info section is present

OlenaReukova committed:
test: Ensure the copyright notice is visible

nichgalzin reviewed (changes requested):
A few notes:
- Mobile version of this looks great but is missing the links for the "privacy policy" and and "term and conditions"
- The spacing and padding  looks a bit strange when this is one a tablet sized device:
![Screenshot 2025-03-24 at 11 07 36](https://github.com/user-attachments/assets/5b85529a-67d2-4cdc-8a0a-74b8250bd24c)
- On desktop the padding need to have 100px gutter on either side of the text (let me know if this is not clear)
- We don't want the footer on the conversation page, but it currently is there

OlenaReukova commented:
> A few notes:
> 
> * Mobile version of this looks great but is missing the links for the "privacy policy" and and "term and conditions"
> * The spacing and padding  looks a bit strange when this is one a tablet sized device:
>   ![Screenshot 2025-03-24 at 11 07 36](https://private-user-images.githubusercontent.com/74174344/426049159-5b85529a-67d2-4cdc-8a0a-74b8250bd24c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI4MzU3MzYsIm5iZiI6MTc0MjgzNTQzNiwicGF0aCI6Ii83NDE3NDM0NC80MjYwNDkxNTktNWI4NTUyOWEtNjdkMi00Y2RjLThhMGEtNzRiODI1MGJkMjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI0VDE2NTcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1M2M1NGQwY2NlN2FiMzE0OWIyZTQ2NDE0N2E2MDM3YjM0YzM2MjE0ZWIzNmY5Nzg4M2E0NDM5YzA2MGQ1MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zxpawkwnlmdhx87-22wvaJsSiruBlSkLPouVi3Bukmo)
> * On desktop the padding need to have 100px gutter on either side of the text (let me know if this is not clear)
> * We don't want the footer on the conversation page, but it currently is there

_______________
@nichgalzin, thank you for your time and the code review, I am starting to fix it now.
1) the links for the "privacy policy" and "terms and conditions" will be provided by the UI/UX designers in the last stage. 
These pages are not in the prior list for them now.

2) Fixed the tablet's responsiveness - 4 columns.

3) On the desktop the padding is 100px gutter on either side of the text. 

4) Added DynamicFooter logic to exclude the Footer component from /conversations.

OlenaReukova committed:
fix: tablet responsiveness

OlenaReukova committed:
fix: add 100px gutter on either side of the text

OlenaReukova committed:
fix: tablet's responsiveness

OlenaReukova committed:
add: DynamicFooter logic to exclude the Footer component from /conversations.

OlenaReukova committed:
fix: desktop 100px gutter on either side of the text

OlenaReukova committed:
fix: On large screens, decreases gap to 16px.

nichgalzin reviewed (approved):
Great work Olena. This all looks great.
